tmol/io/canonical_ordering.py

```diff
@@ -335,14 +383,19 @@ def _d(x):
         )
 
 
+@validate_args
 @toolz.functoolz.memoize
```

Comment:
does validate_args do anything on functions w no arguments?  Does it validate return type?